models/DataObject/ClassDefinition/Data.php

```diff
@@ -1199,6 +1199,104 @@ public function getUnique()
         return false;
     }
 
+    /**
+     * @param mixed $data
+     * @param DataObject\Concrete|DataObject\Localizedfield|DataObject\Objectbrick\Data\AbstractData|DataObject\Fieldcollection\Data\AbstractData $object
+     * @param array $params
+     *
+     * @throws \Exception
```

Comment:
@throws Exception ... wondering in which case

Reply:
If relations are filtered then it calls `setDataToObject` which tries to set the data back to object so could be any scenario like `setLocalizedValue` fails etc.